source/ConditionsStore.cpp

```diff
@@ -399,6 +405,58 @@ int64_t ConditionsStore::PrimariesSize() const
 
 
 
+// Expands the name of a condition to the full name based on the $[] blocks it contains.
+string ConditionsStore::Expand(const string &name) const
+{
+	using size_t = string::size_type;
+	size_t it = name.find("$[");
+
+	if(it == string::npos)
+		return name;
+
+	const size_t len = name.length();
+	string builder;
+	size_t end = -1;
+
+	while(it != string::npos)
+	{
+		builder.append(name.substr(end + 1, it - end - 1)); // does not append the iterator char
+		int depth = 1;
+
+		bool valid = true;
+		for(size_t i = it + 2;; i++)
+		{
+			if(i == len)
+			{
+				valid = false;
+				break;
+			}
+			if(i < len - 1 && name[i] == '$' && name[i + 1] == '[')
+			{
+				depth++;
+				i++;
+			}
+			else if(name[i] == ']')
+			{
+				depth--;
+				if(depth == 0)
+				{
+					end = i;
+					break;
+				}
+			}
+		}
+		if(!valid)
+			break;
+		builder.append(to_string(Get(name.substr(it + 2, end - it - 2))));
+		it = name.find("$[", end + 1);
+	}
+	builder.append(name.substr(end + 1, len - end));
+	return builder;
+}
+
+
+
 ConditionsStore::ConditionEntry *ConditionsStore::GetEntry(const string &name)
```

Comment:
it isnt in the same order as the hpp, is it?

Reply:
What do you mean?

Reply:
getentry is earlier in the hpp

Reply:
Ah, so I should move Expand below that? Sure.

Reply:
yeah below verifyproviderlocation